docs/admin/etcd.md

```diff
@@ -1,49 +1,203 @@
 ---
 assignees:
-- lavalamp
-title: Configuring Kubernetes Use of etcd
+- mml
+- wojtek-t
+title: Upgrading and Rolling Back etcd
 ---
 
+## About etcd
+
 [etcd](https://coreos.com/etcd/docs/latest/) is a highly-available key value
 store which Kubernetes uses for persistent storage of all of its REST API
-objects.
+objects.  For the mechanics behind how kubernetes builds, distributes and
+deploys etcd, see _some doc_.
 
-## Configuration: high-level goals
+### etcd for Kubernetes: high-level goals
 
 Access Control: give *only* kube-apiserver read/write access to etcd. You do not
 want apiserver's etcd exposed to every node in your cluster (or worse, to the
 internet at large), because access to etcd is equivalent to root in your
 cluster.
 
 Data Reliability: for reasonable safety, either etcd needs to be run as a
-[cluster](/docs/admin/high-availability/#clustering-etcd) (multiple machines each running
-etcd) or etcd's data directory should be located on durable storage (e.g., GCE's
-persistent disk). In either case, if high availability is required--as it might
-be in a production cluster--the data directory ought to be [backed up
-periodically](https://coreos.com/etcd/docs/latest/op-guide/recovery.html),
+[cluster](/docs/admin/high-availability/#clustering-etcd) (multiple machines
+each running etcd) or etcd's data directory should be located on durable storage
+(e.g., GCE's persistent disk). In either case, if high availability is
+required--as it might be in a production cluster--the data directory ought to be
+[backed up periodically](https://coreos.com/etcd/docs/latest/op-guide/recovery.html),
 to reduce downtime in case of corruption.
 
-## Default configuration
 
-The default setup scripts use kubelet's file-based static pods feature to run etcd in a
-[pod](http://releases.k8s.io/{{page.githubbranch}}/cluster/saltbase/salt/etcd/etcd.manifest). This manifest should only
-be run on master VMs. The default location that kubelet scans for manifests is
-`/etc/kubernetes/manifests/`.
+## Background
+
+As of 1.5.1 kubernetes release, we are still using etcd from 2.2.1 release with
+v2 API.  Also, we have no pre-existing process for updating etcd, as we have
+never updated etcd by either minor or major version.
+
+Note that we need to migrate both: etcd versions that we are using (from 2.2.1
+to at least 3.0.x) as well as the version of API Kubernetes talks to etcd (3.0.x
+etcd binaries support both v2 and v3 API).
+
+This document describes how to do this migration.
+
+### etcd upgrade limitations
+
+The way etcd was designed introduces some significant limitations on how the
```

Comment:
> The way etcd was designed introduces some significant limitations on how the upgrade can be done. These are the main limitations.

I disagree with the wording here. It sounds more like complaints than clarifying the limits to users.

Maybe rephrase like:
```
There are some limits in etcd upgrade process.
```